app/controllers/admin/users_controller.rb

```diff
@@ -13,6 +13,27 @@ def index
       end
     end
 
+    def new
+      @user = User.new
+    end
+
+    def create
+      user_params = params.require(:user)
+      user_type = user_params[:type].camelize.downcase
+      case user_type
+      when 'enduser'
+        user_params[:type] = 'EndUser'
+      when 'adminuser'
+        user_params[:type] = 'AdminUser'
+      else
+        user_params[:type] = nil
+      end
+      @user = User.create(user_params.permit(*DEFAULT_FIELDS))
+      respond_with @user do |format|
+        format.html { redirect_to admin_users_path }
```

Comment:
Overriding the response format like this seems to resolve the failing tests. Not sure why.

Reply:
Resolved. It took embarrassingly too long, but after leaving and coming back to this problem I finally realize tests failed due to errors with the test itself as opposed to the code.